tests/e2e/vsphere_volume_expansion_staging.go

```diff
@@ -0,0 +1,1331 @@
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+	fss "k8s.io/kubernetes/test/e2e/framework/statefulset"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+)
+
+var _ = ginkgo.Describe("[csi-supervisor-staging] Tests for WCP env with minimal permission user", func() {
+
+	var (
+		client            clientset.Interface
+		namespace         string
+		storagePolicyName string
+	)
+
+	ginkgo.BeforeEach(func() {
+		var err error
+		if k8senvsv := GetAndExpectStringEnvVar("SUPERVISOR_CLUSTER_KUBE_CONFIG"); k8senvsv != "" {
+			client, err = createKubernetesClientFromConfig(k8senvsv)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		namespace = os.Getenv("SVC_NAMESPACE")
+
+		bootstrap()
+		_, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		nodeList, err := fnodes.GetReadySchedulableNodes(client)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+	})
+
+	/*
+		Verify online volume expansion on dynamic volume
+
+		1. Create StorageClass with allowVolumeExpansion set to true.
+		2. Create PVC which uses the StorageClass created in step 1.
+		3. Wait for PV to be provisioned.
+		4. Wait for PVC's status to become Bound and note down the size
+		5. Create a Pod using the above created PVC
+		6. Modify PVC's size to trigger online volume expansion
+		7. verify the PVC status will change to "FilesystemResizePending". Wait till the status is removed
+		8. Verify the resized PVC by doing CNS query
+		9. Make sure data is intact on the PV mounted on the pod
+		10.  Make sure file system has increased
+
+	*/
+	ginkgo.It("Verify online volume expansion on staging dynamic volume", func() {
+		ginkgo.By("Verify online volume expansion on staging dynamic volume")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var pvclaim *v1.PersistentVolumeClaim
+		var pv *v1.PersistentVolume
+
+		ginkgo.By("Creating a PVC")
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pvclaim, err = fpv.CreatePVC(client, namespace,
+			getPersistentVolumeClaimSpecWithStorageClass(namespace, diskSize, storageclass, nil, v1.ReadWriteOnce))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client,
+			[]*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+		pv = persistentvolumes[0]
+
+		defer func() {
```

Comment:
Add this defer , Before checking bound.
Suggesting this because , If in any case suppose PVC wont get into bound state and test fails. The Pvc will remain in Pending state without getting deleted.  
Or add an after each and handle this condetion. 
 Just to be sure since this is staging testbed